src/h3/mod.rs

```diff
@@ -26,4 +26,940 @@
+) -> http::Request<()> {
+    let mut req: Request<()> = Request::default();
+
+    // TODO: make pseudo header parsing more efficient. Right now, we create
+    // some variables to hold pseudo headers that may arrive in any order.
+    // Some of these are later formatted back into a complete URI
+    let mut method = String::new();
+    let mut scheme = String::new();
+    let mut authority = String::new();
+    let mut path = String::new();
+
+    for hdr in decoder.decode(hdr_block).unwrap() {
+        // trace!("Header field - {}:{}", hdr.0, hdr.1);
+
+        match hdr.name() {
+            ":method" => {
+                method = hdr.value().to_string();
+            },
+            ":scheme" => {
+                scheme = hdr.value().to_string();
+            },
+            ":authority" => {
+                authority = hdr.value().to_string();
+            },
+            ":path" => {
+                path = hdr.value().to_string();
+            },
+            _ => {
+                req.headers_mut().insert(
+                    http::header::HeaderName::from_bytes(hdr.name().as_bytes())
+                        .unwrap(),
+                    http::header::HeaderValue::from_str(hdr.value()).unwrap(),
+                );
+            },
+        }
+    }
+
+    let uri = format!("{}://{}{}", scheme, authority, path);
+
+    *req.method_mut() = method.parse().unwrap();
+    *req.version_mut() = http::Version::HTTP_2;
+    *req.uri_mut() = uri.parse::<Uri>().unwrap();
+
+    // debug!("Prepared request {:?}", req);
+
+    req
+}
+
+fn resp_hdrs_from_qpack(
+    decoder: &mut qpack::Decoder, hdr_block: &mut [u8],
+) -> http::Response<()> {
+    let mut resp: Response<()> = Response::default();
+
+    // TODO: make pseudo header parsing more efficient.
+    let mut status = String::new();
+
+    for hdr in decoder.decode(hdr_block).unwrap() {
+        // trace!("Header field - {}:{}", hdr.0, hdr.1);
+
+        match hdr.name() {
+            ":status" => {
+                status = hdr.value().to_string();
+            },
+            _ => {
+                resp.headers_mut().insert(
+                    http::header::HeaderName::from_bytes(hdr.name().as_bytes())
+                        .unwrap(),
+                    http::header::HeaderValue::from_str(hdr.value()).unwrap(),
+                );
+            },
+        }
+    }
+
+    *resp.status_mut() = StatusCode::from_bytes(status.as_bytes()).unwrap();
+    *resp.version_mut() = http::Version::HTTP_2;
+
+    // debug!("Prepared response {:?}", resp);
+
+    resp
+}
+
+/// An HTTP/3 configuration.
+pub struct Config {
+    pub root_dir: String,
+    pub num_placeholders: u64,
+    pub max_header_list_size: u64,
+    pub qpack_max_table_capacity: u64,
+    pub qpack_blocked_streams: u64,
+}
+
+impl Config {
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new() -> Result<Config> {
+        Ok(Config {
+            root_dir: String::new(),
```

Comment:
AFAICT this field is never used, can we just remove it? I don't think it's a good idea to have the library handle this anyway.

Reply:
Removed and now used by example http3-server to actually serve files!